packages/hash/frontend/src/components/layout/PageHeader/AccountDropdown.tsx

```diff
@@ -25,19 +24,27 @@ export const AccountDropdown: VoidFunctionComponent<AccountDropdownProps> = ({
   logout,
   user,
 }) => {
-  const theme = useTheme();
-
   const buttonRef = useRef(null);
 
   const [open, setOpen] = useState(false);
 
   const id = open ? "account-popover" : undefined;
 
+  const userPrimaryEmail = useMemo(() => {
+    const primaryEmail = user.properties.emails.find((email) => email.primary);
+
+    if (!primaryEmail) {
+      throw new Error("User doesn't have a primary email set.");
```

Comment:
I don't think it's a good idea to through an error in this case especially since we are not handling it. Better to display a fallback value instead

Reply:
Have a look at https://github.com/hashintel/hash/pull/397#discussion_r825459716

If you disagree with my thinking, happy for this to be reverted - I don't have strong opinions on this. Just feel we are creating more opportunities for something critical to slip through.

I'd rather have this function return `undefined` and the `AccountDropdown` display no email, instead of the incorrect email.

Reply:
@benwerner01 I missed the earlier comment, thanks for pointing me to it. I agree the AccountDropdown should probably display no email in that scenario instead of using a random fallback. Although I disagree with the error. If our FE tests don't catch the error and a user/users end up having the problem, they will be unable to use the app. I don't see why the app should break because of this. We can instead log the issue with Sentry

Reply:
And apologies @akash-joshi for leading you down the wrong path in terms of how to implement this, the change should hopefully be as simple as adding `return undefined` instead of throwing the error.

Reply:
Done here: https://github.com/hashintel/hash/pull/397/commits/9a73141ef5bdbecec289ef1263792825c4c6e2f5